### PR TITLE
Verify schema version

### PR DESCRIPTION
**Pull request by rzezeski**
This patch verifies that the schema version is set to `1.5` and is the
second part of enhancements for issues #321. There was also
supplemental work done around error messages so that they would
display in a more user friendly manner.

Result of uploading schema with wrong version:

```
# curl -i -X PUT -H 'content-type: application/xml' 'http://localhost:10018/search/schema/bad_version' --data-binary @/root/bad-version-schema.xml
HTTP/1.1 100 Continue

HTTP/1.1 400 Bad Request
Server: MochiWeb/1.1 WebMachine/1.10.5 (jokes are better explained)
Date: Tue, 25 Feb 2014 21:08:06 GMT
Content-Type: text/plain
Content-Length: 61

Error storing schema: schema 'version' attribute must be 1.5
```

Attempting to store schema with missing fields (yes, the x-path is
ugly but this is still an improvement from raw erlang terms):

```
# curl -i -X PUT -H 'content-type: application/xml' 'http://localhost:10018/search/schema/bad_version' --data-binary @/root/missing-fields-schema.xml
HTTP/1.1 100 Continue

HTTP/1.1 400 Bad Request
Server: MochiWeb/1.1 WebMachine/1.10.5 (jokes are better explained)
Date: Tue, 25 Feb 2014 21:08:59 GMT
Content-Type: text/plain
Content-Length: 205

Error storing schema: missing field /schema/fields/field[@name="_yz_fpn" and @type="_yz_str" and @indexed="true"]
missing field /schema/fields/field[@name="_yz_pn" and @type="_yz_str" and @indexed="true"]
```

Attempting to store a truncated schema:

```
# curl -i -X PUT -H 'content-type: application/xml' 'http://localhost:10018/search/schema/bad_version' --data-binary @/root/truncated-schema.xml
HTTP/1.1 100 Continue

HTTP/1.1 400 Bad Request
Server: MochiWeb/1.1 WebMachine/1.10.5 (jokes are better explained)
Date: Tue, 25 Feb 2014 21:10:52 GMT
Content-Type: text/plain
Content-Length: 203

Error storing schema: failed to parse {fatal,
                    {unexpected_end,
                        {file,file_name_unknown},
                        {line,161},
                        {col,1}}}
```

### CODE REVIEW

**Comment by bowrocker** [comment]
- [x] compile/eunit: https://gist.github.com/bowrocker/9248817
- [ ] dialyzer: broken at the moment
- [x] riak tests: https://gist.github.com/bowrocker/9239243
- [x] code review

I tried various bad values for the 'version' field, and consistently got the error message:

```
->  curl -i -X PUT -H 'content-type: application/xml' 'http://localhost:10018/search/schema/books' --data-binary @/Users/janderson/sandbox/develop/riak/dev/dev1/book.xml
HTTP/1.1 100 Continue

HTTP/1.1 400 Bad Request
Server: MochiWeb/1.1 WebMachine/1.10.5 (jokes are better explained)
Date: Thu, 27 Feb 2014 12:34:17 GMT
Content-Type: text/plain
Content-Length: 61

Error storing schema: schema 'version' attribute must be 1.5
```

:+1:  018d960
:+1:  a229fcb

File: src/yz_schema.erl [hunk 1]
```diff
@@ -91,19 +95,35 @@ parse_and_verify(RawSchema) ->
...
-    verify_fts(verify_fields(verify_uk(Schema))).
+    verify_fts(verify_fields(verify_vsn(verify_uk(Schema)))).
+
+%% @private
+%%
+%% @doc Verify the the schema 'version' attribute is set to correct
+%% value.
+-spec verify_vsn({ok, schema()} | schema_err()) -> {ok, schema()} | schema_err().
+verify_vsn({ok, Schema}) ->
+    case xmerl_xpath:string("string(/schema/@version)", Schema) of
```
Comment (coderoshi) [inline src/yz_schema.erl:114]: The other xpaths defines are in yokozuna.hrl, how about this one too?

**Comment by coderoshi** [comment]
one style comment, +1 on code and test.

One unrelated AAE test did fail for me, however, which I'll post here for completeness.

```
================ aae_test failure stack trace =====================
{{assertEqual_failed,[{module,aae_test},
                      {line,273},
                      {expression,"RepairCount"},
                      {expected,83},
                      {value,79}]},
 [{aae_test,'-verify_repair_count/2-fun-0-',2,
            [{file,"riak_test/aae_test.erl"},{line,273}]},
  {aae_test,verify_repair_count,2,
            [{file,"riak_test/aae_test.erl"},{line,273}]},
  {aae_test,confirm,0,[{file,"riak_test/aae_test.erl"},{line,69}]}]}
===================================================================
```

File: src/yz_schema.erl (hunk 1 above)
Comment (rzezeski) [inline src/yz_schema.erl:114]: That section is for the custom fields Yokozuna adds. This isn't one of them.

**Comment by rzezeski** [comment]
@bowrocker This branch should have 5 or 6 dialyzer errors as it was before the dialyzer fixes of late.

@coderoshi I have also seen the AAE test fail in a similar manner in the past. I believe there is a race condition because if I leave the cluster up and check the cluster the query results come back correct. The changes in this PR affect nothing in AAE land and so I'm inclined to not hold this patch up for that failure.

Are those answers enough to get me two official +1's?

**Comment by bowrocker** [comment]
@rzezeski yes, :+1:

**Comment by coderoshi** [comment]
yes +1

**Pull request merged**